hicommonwealth/commonwealth
# Fetch Proposal metadata - Cosmos v1

## Link to Issue
Closes: #4184

## Description of Changes
- Adds useProposalMetadata hook for Cosmos V1
- lazy-loads metadata if v1 proposal.metadata IRI is present
- updates proposal in store after lazy-load
- uses IPFS proxy if ipfs file. Non-IPFS resources are disabled for now.

## Test Plan
- Added tests for proposals.spec E2e
- CA (click around) tested on local:
  - http://localhost:8080/kyve/proposals -> expect titles populated for Props 5 and 6
  - click to Prop 5 or 6 -> expect title and description populated
  - click to another community, paste and go directly to http://localhost:8080/kyve/proposal/6 -> expect title and description to load

## Other Considerations

- Follow-up to enable non-IPFS: https://github.com/hicommonwealth/commonwealth/issues/4233
